yt_dlp/extractor/youtube.py

```diff
@@ -611,20 +603,16 @@ def _extract_continuation(cls, renderer):
             if continuation:
                 return continuation
 
-    @staticmethod
-    def _extract_alerts(data):
+    @classmethod
+    def _extract_alerts(cls, data):
         for alert_dict in try_get(data, lambda x: x['alerts'], list) or []:
             if not isinstance(alert_dict, dict):
                 continue
             for alert in alert_dict.values():
                 alert_type = alert.get('type')
                 if not alert_type:
                     continue
-                message = try_get(alert, lambda x: x['text']['simpleText'], compat_str) or ''
-                if message:
-                    yield alert_type, message
-                for run in try_get(alert, lambda x: x['text']['runs'], list) or []:
-                    message += try_get(run, lambda x: x['text'], compat_str)
+                message = cls._get_text(alert.get('text'))
                 if message:
                     yield alert_type, message
```

Comment:
potential future bug: if we fail to extract a message this will not return a two item tuple. it is assumed it will in other functions

Reply:
we are yielding the values, not returning it. So the iterator will simply be empty if we fail to extract message